src/pybind/mgr/dashboard/controllers/smb.py

```diff
@@ -87,6 +88,19 @@
     }, "Configuration for the CephFS share")
 }
 
+def raise_unsuccess(func):
+    @wraps(func)
+    def wrapper(*args, **kwargs):
+        result = func(*args, **kwargs)
+
+        if isinstance(result, dict) and 'success' in result and result['success'] is False:
+            if not 'msg' in result:
+                result["msg"] = 'Operation was unsuccessful'
+            raise DashboardException(msg=result['msg'], component='smb')
```

Comment:
```suggestion
            msg = result["msg"] if "msg" in result else 'Operation was unsuccessful'
            raise DashboardException(msg=msg, component='smb')
```
No need to mutate result if you're just about to discard it.

Reply:
Got it. Thanks for the pointer.

I've updated it.